[Security Solution][Detection Engine] EUI Tech Debt - Removes references to static EuiTheme variables

rylnd opened the pull request:
## Summary

This PR is a followup to #205990, which removed references to all of the deprecated/renamed EUI vars in preparation for 9.0. Here, we address some of the non-critical tech debt related to the EUI refresh, namely the [removal of static EUI tokens](https://github.com/elastic/kibana/issues/199715#json-tokens) from our codebase.

I made every attempt not to change any styles in this PR, except to simplify CSS to produce an equivalent design. A common  example of this was removing a static `margin` or `padding` declaration referencing `euiThemeVars.size*`, and swapping it with an equivalent `gutterSize` prop on the `EuiFlexGroup` container, or with an `align-self` or other equivalent flexbox directive.

## Screenshots of Areas Affected
The majority of changes here involved the Exception List/Item pages. I've attached screenshots of their current layout for comparison/review:


<details>
  <summary>
    <h3>Rule Exceptions Tab</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="1395" alt="Rule exceptions tab - before" src="https://github.com/user-attachments/assets/db7a5487-7df3-4a5f-b88e-90ab34784970" />

  </kbd>
  <kbd>
    <h3>After</h3>
<img width="1421" alt="Rule exceptions tab - after" src="https://github.com/user-attachments/assets/77cbdefc-cbec-4b9e-8436-197f2f2f6677" />

  </kbd>
</details>

<details>
  <summary>
    <h3>Shared Exception Lists</h3>
  </summary>
  <kbd>
<img width="1517" alt="Shared Exception Lists" src="https://github.com/user-attachments/assets/5448dd65-60f7-470c-bf7a-7af75bb914fa" />

  </kbd>
</details>

<details>
  <summary>
    <h3>Shared Exception List Details</h3>
  </summary>
  <kbd>
<img width="1517" alt="Shared Exception List Details" src="https://github.com/user-attachments/assets/6e03fef6-af58-40bb-98c5-188651a584cc" />

  </kbd>
</details>

<details>
  <summary>
    <h3>Threshold Input</h3>
  </summary>
  <kbd>
<img width="1046" alt="Threshold Input" src="https://github.com/user-attachments/assets/28738857-6bdf-404f-a790-a9f4e66ff27a" />

  </kbd>
</details>



### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.


- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

rylnd committed:
Fix incorrect DOM nesting on ExceptionListHeader component

The EUI page header has some parent <p> somewhere, and we were
incorrectly putting <div>s as children. This fixes that, along with a
few necessary style changes (and updated snapshots).

rylnd committed:
Remove euiThemeVars from Empty View component

rylnd committed:
Make styles Exception Item Expression in dynamic to EUI theme

This replaces static references to euiThemeVars to enable this behavior.

I did away with some `margin` usage in favor of `EuiFlexGroup`'s
`gutterSize` prop, as well as removing some manual `display: flex`s in
favor of EuiFlexItem/Group.

This also appears to fix some errors in the snapshot test related to
this component. I'm not sure how those happened, but it appears specific
to that test and not a production behavior.

rylnd committed:
Make font-family styles dynamic with theme

rylnd committed:
Remove unnecessary manual styles from Shared Exception Details

While removing static references to euiThemeVars, I found that these
manual styles weren't needed:

* `&div:first-child` selector wasn't being applied
* margins can be replaced with a different `gutterSize` prop
* the eui-yScrollWithShadow was adding a shadow to a container that
  wasn't actually scrollable; when the extra margin above was removed,
  this caused the shadow to cover part of the border of the Exception
  item panels

rylnd committed:
Port Exception Menu Link styles to be dynamic

This ports the styles over from static generation with euiThemeVars to
dynamic with useEuiTheme. I also simplified styles in a few ways, here:

1. Remove the JS conditions around applying styles, and porting to
   equivalent CSS
2. Removing unnecessary flex-basis CSS (which didn't work on the div it
   was being applied to)
3. Moving child `color` CSS to the parent, instead.

rylnd committed:
Replace custom CSS with appropriate `gutterSize` prop

We had negative-margin CSS to offset a 24px `gap` on our flex grouping.
By reducing the `gap` to 8px, we no longer need to offset it.

rylnd committed:
Replace padding style with another `gutterSize` prop

Instead of adding bottom padding to each element in a container (the
header of an individual exception list on its Shared Details page), we
instead make the container a `FlexGroup` and add an analogous
`gutterSize` prop (and `margin-bottom`) to it.

rylnd committed:
Update snapshots

rylnd committed:
Fix DOM nesting issue by using span for flexgroup

I forgot about the aforementioned weirdness in EUIPageHeader that
necessitates everything being a span.

rylnd committed:
Remove unnecessary styles

We were wrapping a `span` in a `div`, and then giving that div:

1. display: inline
2. margin-left: 4px

Because the margin is so insignificant (there is already natural space
after the previous element), I opted to just remove the div and its
styles altogether.

rylnd committed:
Convert MlJobItem to use @emotion and dynamic vars

The generated styles are equivalent.

rylnd committed:
Replace use of styled-components/euiThemevars with @emotion/styled

rylnd committed:
Remove usage of styled-components and EuiThemeVars from ExceptionsListCard

This component exists as the inner item on the Shared Exception Lists
page. Styles have been verified as equivalent.

rylnd committed:
Replace use of static theme vars via @emotion/styled

rylnd committed:
Remove use of static theme vars in ValueListModal body

rylnd committed:
Remove static theme vars from Threshold Rule Creation components

* Defines a FieldSectionGroup component instead of applying the same
  styles to multiple identical EuiFieldGroup components
* Replaces use of margin-top with a more appropriate `align-self`
  declaration

rylnd commented:
/ci

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/271730)
* Commit: 61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/271730#0194b349-ea8f-4440-b7f1-65a72f66f7de)

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271730","number":271730,"commit":"61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903"}],"number":271730}
buildkite-pr-comment-->

rylnd committed:
Fix styles on "Linked Rules" link in Rule Exceptions Tab

These were already fixed on the package version of this component, but
the one that's still in the plugin was not updated, which I noticed
while taking screenshots.

rylnd commented:
Note to self: I think that [7b5441e](https://github.com/elastic/kibana/pull/208820/commits/7b5441e5fbc8fd11a3f220707145ca8e19f198cf) was a mistake. That function should reference _components_ that use the `useEuiTheme()` hook, as opposed to calling the hook directly.

rylnd committed:
Add EuiTheme to @emotion types in exception-list package

rylnd committed:
Fix/update tests following new use of EuiTheme

These tests were failing because they lacked the EUI Theme context when
we were rendering these components which now require them (previously
they were requiring a static object imported from elsewhere).

rylnd commented:
/ci

rylnd committed:
Use clarifying testing syntax

We don't really want to test the EuiThemeProvider, it's merely a
consequence of the component's dependencies/how it's written. Calling
the provider out as a wrapper better helps to show that relationship.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/272469)
* Commit: eb60210facbea050b52f98ff2ec8ef5d3a55ebab

### Failed CI Steps
* [Defend Workflows Cypress Tests on Serverless #7](https://buildkite.com/elastic/kibana-pull-request/builds/272469#0194bec4-be1f-479a-bb44-398f3c4486a6)

### History
* :broken_heart: [Build #271730](https://buildkite.com/elastic/kibana-pull-request/builds/271730) failed 61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272469","number":272469,"commit":"eb60210facbea050b52f98ff2ec8ef5d3a55ebab"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271730","number":271730,"commit":"61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903"}],"number":272469}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/eui_tech_debt

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/272489)
* Commit: eca40cd9718d8927720a1044b9f6286d0bbb07d0

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/272489#0194bfc3-f159-496f-9dc4-d90a217ee799)

### History
* :broken_heart: [Build #272469](https://buildkite.com/elastic/kibana-pull-request/builds/272469) failed eb60210facbea050b52f98ff2ec8ef5d3a55ebab
* :broken_heart: [Build #271730](https://buildkite.com/elastic/kibana-pull-request/builds/271730) failed 61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272489","number":272489,"commit":"eca40cd9718d8927720a1044b9f6286d0bbb07d0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272469","number":272469,"commit":"eb60210facbea050b52f98ff2ec8ef5d3a55ebab"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271730","number":271730,"commit":"61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903"}],"number":272489}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/eui_tech_debt

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/272507)
* Commit: 7684a34a7f5d5bc411d35706bc4f57d3cfaf5ef6

### Failed CI Steps
* [FTR Configs #17](https://buildkite.com/elastic/kibana-pull-request/builds/272507#0194c2cd-1c1e-47cc-9fe9-551586a379fd)
* [FTR Configs #17](https://buildkite.com/elastic/kibana-pull-request/builds/272507#0194c2f4-75ce-4819-bb6d-5b85a94f19d9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/272507#0194c2cd-1c1e-47cc-9fe9-551586a379fd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/272507/jobs/0194c2cd-1c1e-47cc-9fe9-551586a379fd/artifacts/0194c2f4-3c6d-480e-9ad9-60f6ab2d4e37) FTR Configs #<span></span>17 / Fleet Endpoints Integrations inputs_with_standalone_docker_agent "before all" hook for "generate a valid config for standalone agents"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/272507#0194c2f4-75ce-4819-bb6d-5b85a94f19d9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/272507/jobs/0194c2f4-75ce-4819-bb6d-5b85a94f19d9/artifacts/0194c306-a6ce-44a4-8d76-efc8a52b2ecf) FTR Configs #<span></span>17 / Fleet Endpoints Integrations inputs_with_standalone_docker_agent "before all" hook for "generate a valid config for standalone agents"

### History
* :broken_heart: [Build #272489](https://buildkite.com/elastic/kibana-pull-request/builds/272489) failed eca40cd9718d8927720a1044b9f6286d0bbb07d0
* :broken_heart: [Build #272469](https://buildkite.com/elastic/kibana-pull-request/builds/272469) failed eb60210facbea050b52f98ff2ec8ef5d3a55ebab
* :broken_heart: [Build #271730](https://buildkite.com/elastic/kibana-pull-request/builds/271730) failed 61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272507","number":272507,"commit":"7684a34a7f5d5bc411d35706bc4f57d3cfaf5ef6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272489","number":272489,"commit":"eca40cd9718d8927720a1044b9f6286d0bbb07d0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272469","number":272469,"commit":"eb60210facbea050b52f98ff2ec8ef5d3a55ebab"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271730","number":271730,"commit":"61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903"}],"number":272507}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/eui_tech_debt

rylnd commented on x-pack/solutions/security/plugins/security_solution/public/value_list/components/value_list_modal.tsx:
```diff
@@ -6,6 +6,7 @@
  */
 import React, { useState, useCallback } from 'react';
 import { css } from '@emotion/css';
+import styled from '@emotion/styled';
```
@vitaliidm this is not the incompatible `styled-components` package but rather a variant of emotion's syntax. It's mentioned later in the comment you referenced: 

> As a simple first step in this transition, consider using [@emotion/styled](https://emotion.sh/docs/styled). This allows you to keep the familiar styled syntax while aligning with the Emotion ecosystem

It's unclear whether we'll need to change these in the future, but if so we can address them similarly/concurrently with the removal of remaining `styled-components` references.

vitaliidm replied:
Yes, it's part of`@emotion`, but what got me curious - why the need to introduce `styled` in this file and couple of others?

We did not have `styled-components` here at all, so using it `@emotion/styled` as a transition step does not apply in this case.
In `x-pack/solutions/security/plugins/security_solution/public/exceptions/pages/shared_lists/index.tsx`  for example, `styled-components` were replaced by `@emotion/styled` and I see why you did it there. It's in line with the eui-team guide.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/273311)
* Commit: 3df4d2d51397cbc4179c1cae20938224a73da64a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/6f55501a75c215f7e802a516d051ceb44df63a03) | [after](https://github.com/elastic/kibana/commit/3df4d2d51397cbc4179c1cae20938224a73da64a) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 6656 | 6654 | -2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6f55501a75c215f7e802a516d051ceb44df63a03) | [after](https://github.com/elastic/kibana/commit/3df4d2d51397cbc4179c1cae20938224a73da64a) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 21.5MB | 21.5MB | -3.8KB |



### History
* :broken_heart: [Build #272507](https://buildkite.com/elastic/kibana-pull-request/builds/272507) failed 7684a34a7f5d5bc411d35706bc4f57d3cfaf5ef6
* :broken_heart: [Build #272489](https://buildkite.com/elastic/kibana-pull-request/builds/272489) failed eca40cd9718d8927720a1044b9f6286d0bbb07d0
* :broken_heart: [Build #272469](https://buildkite.com/elastic/kibana-pull-request/builds/272469) failed eb60210facbea050b52f98ff2ec8ef5d3a55ebab
* :broken_heart: [Build #271730](https://buildkite.com/elastic/kibana-pull-request/builds/271730) failed 61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/273311","number":273311,"commit":"3df4d2d51397cbc4179c1cae20938224a73da64a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272507","number":272507,"commit":"7684a34a7f5d5bc411d35706bc4f57d3cfaf5ef6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272489","number":272489,"commit":"eca40cd9718d8927720a1044b9f6286d0bbb07d0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/272469","number":272469,"commit":"eb60210facbea050b52f98ff2ec8ef5d3a55ebab"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271730","number":271730,"commit":"61f81a2b9c7bea80bb7e3b7fd49f972a1fc4b903"}],"number":273311}
buildkite-pr-comment-->

vitaliidm reviewed (approved):
Changes LGTM. Thanks for addressing tech debt
